### ISSUE TITLE
Add support for git tags and other commits for funcbench in GitHub Mode

### ISSUE THREAD

**Issue by geekodour** [prometheus/test-infra#373]
Currently funcbench only supports branch name and commit hash as the target. It'll be nice to add support to[ comment like in this comment.](https://github.com/prometheus/prometheus/pull/7287#issuecomment-638246346)

`/funcbench <branch> <benchmark function regex>`

I also think when running funcbench in Github Mode it only allows comparing the PR to the master and not to even any other commit or branch because it clones the repo with depth1. (See logs below)

https://github.com/prometheus/test-infra/blob/3d826afdf0a60ae8440ca8681561ce8240805ade/funcbench/env.go#L87

```
λ ./funcbench --owner=prometheus-community --nocomment --github-pr="63" 8755434350ba9f8d3e5434826a9fcfb3a1327cb0 BenchmarkBufferedSeriesIterator
Enumerating objects: 26444, done.
Counting objects: 100% (26444/26444), done.
Compressing objects: 100% (16010/16010), done.
Total 26444 (delta 12021), reused 22272 (delta 8718), pack-reused 0
funcbech09:34:01 env.go:127: [GitHub Mode] prometheus-community : prometheus
Benchmarking PR - 63 versus: 8755434350ba9f8d3e5434826a9fcfb3a1327cb0
Benchmark func regex: BenchmarkBufferedSeriesIterator
funcbech09:34:01 main.go:223: Target: 8755434350ba9f8d3e5434826a9fcfb3a1327cb0 Current Ref: 4f8680e6fd14c7289e10a249e1d7128400886cfd
funcbech09:34:01 main.go:239: Assuming comparing with target (clean workdir will be checked.)
funcbech09:34:01 bench.go:88: Executing benchmark command for 4f8680e6fd14c7289e10a249e1d7128400886cfd
 [sh -c cd /tmp/funcbench/prometheus && go test -run ^$ -bench ^BenchmarkBufferedSeriesIterator$ -benchmem -benchtime 1s -timeout 2h0m0s ./...]
funcbech09:34:19 main.go:257: Checking out (in new workdir): /tmp/funcbench/prometheus/_funcbench-cmp commmit 8755434350ba9f8d3e5434826a9fcfb3a1327cb0
funcbech09:34:19 main.go:50: running command failed: checkout 8755434350ba9f8d3e5434826a9fcfb3a1327cb0 in worktree /tmp/funcbench/prometheus/_funcbench-cmp: error: exit status 128; Command out: fatal: invalid reference: 8755434350ba9f8d3e5434826a9fcfb3a1327cb0

λ cd ~/OpenSource/prometheus

λ ./funcbench 8755434350ba9f8d3e5434826a9fcfb3a1327cb0 BenchmarkBufferedSeriesIterator                                                      [master]
funcbech09:35:10 env.go:61: [Local Mode]
Benchmarking current version versus: 8755434350ba9f8d3e5434826a9fcfb3a1327cb0
Benchmark func regex: BenchmarkBufferedSeriesIterator
funcbech09:35:10 main.go:223: Target: 8755434350ba9f8d3e5434826a9fcfb3a1327cb0 Current Ref: 83619aa9ac566313942c624df6ee721f5b521032
funcbech09:35:10 main.go:239: Assuming comparing with target (clean workdir will be checked.)
funcbech09:35:10 bench.go:88: Executing benchmark command for 83619aa9ac566313942c624df6ee721f5b521032
 [sh -c cd /home/geekodour/OpenSource/prometheus && go test -run ^$ -bench ^BenchmarkBufferedSeriesIterator$ -benchmem -benchtime 1s -timeout 2h0m0s ./...]
funcbech09:35:42 main.go:257: Checking out (in new workdir): /home/geekodour/OpenSource/prometheus/_funcbench-cmp commmit 8755434350ba9f8d3e5434826a9fcfb3a1327cb0
funcbech09:35:43 bench.go:88: Executing benchmark command for 8755434350ba9f8d3e5434826a9fcfb3a1327cb0
 [sh -c cd /home/geekodour/OpenSource/prometheus/_funcbench-cmp && go test -run ^$ -bench ^BenchmarkBufferedSeriesIterator$ -benchmem -benchtime 1s -timeout 2h0m0s ./...]
Results:
benchmark                             8755434350ba9f8d3e5434826a9fcfb3a1327cb0 ns/op     new ns/op     delta
BenchmarkBufferedSeriesIterator-8     16.0                                               15.6          -2.50%

benchmark                             8755434350ba9f8d3e5434826a9fcfb3a1327cb0 MB/s     new MB/s           speedup
BenchmarkBufferedSeriesIterator-8     66817637610.15                                    78510541505.52     1.17x

benchmark                             8755434350ba9f8d3e5434826a9fcfb3a1327cb0 allocs     new allocs     delta
BenchmarkBufferedSeriesIterator-8     0                                                   0              +0.00%

benchmark                             8755434350ba9f8d3e5434826a9fcfb3a1327cb0 bytes     new bytes     delta
BenchmarkBufferedSeriesIterator-8     0                                                  0             +0.00%
funcbech09:36:10 main.go:191: exiting

```

cc: @krasi-georgiev  @nevill

**Comment by krasi-georgiev** [comment]
yeah this is something important to fix

**Comment by geekodour** [comment]
@krasi-georgiev This issue is partly resolved after #374 , tried running the same command, works! thanks to @nevill  :) just adding tag support remains now.

**Comment by nevill** [comment]
~~After #377 get merged, we also need to update the regexp in `eventmap.yml`.~~

seems the regexp can capture the tag names correctly.

**Comment by krasi-georgiev** [comment]
ok feel free to close if it is resolved.

**Comment by geekodour** [comment]
Closed by #377
